### DISCUSSION TITLE
`magick mogrify`: How to set the order of commands?

### DISCUSSION THREAD

**Discussion by tobwen** [ImageMagick/ImageMagick#7487] [Help]
### ImageMagick version

7.1.1-35 Q16-HDRI 1bfce2a62

### Operating system, version and so on

Debian Buster, AMD64

### Description

What's the correct way to transform this (_convert_-like) command into a batch command? The order of commands seems to be different in `magick mogrify`.

### `magick`
![image](https://github.com/user-attachments/assets/9bc5522a-6a48-46e3-b864-bc4e24dc02c6)

``` sh
$ magick -monitor test.png -fx '(u.hue > 0) ? 255 : u' -dither None -threshold 75% -compress Fax test.tif
load image[test.png]: 2338 of 2339, 100% complete
FxNew/Image[test.png]: 2338 of 2339, 100% complete
threshold image[test.png]: 2338 of 2339, 100% complete
contrast-stretch image[test.tif]: 2338 of 2339, 100% complete
classify image colors[test.tif]: 2338 of 2339, 100% completee
assign image colors[test.tif]: 2338 of 2339, 100% complete
save image[test.tif]: 2338 of 2339, 100% completeompletee
```

### `magick mogrify`
Optical result is wrong. `-threshold` was executed BEFORE `-fx`, even when stated differently. 
![image](https://github.com/user-attachments/assets/0991a6ea-c5b2-446a-aaf2-18b323c2f79f)

``` sh
$ magick mogrify -monitor -fx '(u.hue > 0) ? 255 : u' -dither None -threshold 75% -compress Fax -format tif '*.png'
load image[test.png]: 2338 of 2339, 100% complete
list
FxNew/Image[test.tif]: 2338 of 2339, 100% complete
contrast-stretch image[test.tif]: 2338 of 2339, 100% complete
classify image colors[test.tif]: 2338 of 2339, 100% complete
assign image colors[test.tif]: 2338 of 2339, 100% complete
save image[test.tif]: 2338 of 2339, 100% completecomplete
```

### workaround

``` sh
find . -maxdepth 1 -name '*.png' -print0 | xargs -0 -P12 -i magick -monitor "{}" -fx "(u.hue > 0) ? 255 : u" -threshold 75% -compress Fax -set filename:fn '%[basename]' '%[filename:fn].tif'
```

**Comment by snibgo** [comment]
`magick mogrify` and `magick convert` use the very old v6 syntax, which executes options in some strange order that is difficult for mere humans to predict. The v7 syntax used by just `magick` (without `mogrify` or `convert`) executes options in the order they are given on the command.
